### PR TITLE
[ACR] add --allow-trusted-services support

### PR DESCRIPTION
**Pull request by adewaleo**
**Description**

This PR adds the option `--allow-trusted-services` to `az acr create` and `az acr update`. By default ACR allows trusted azure services to access network restricted registries.

**Testing Guide**
```
az acr create -h 
az acr update -h
azdev test test_acr_with_public_network_access
```
**History Notes**

[ACR] az acr create / update: add --allow-trusted-services. This parameter determines whether trusted azure services are allowed to access network restricted registries. The default is to allow.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -82,7 +82,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         with self.argument_context(scope, arg_group='Network Rule') as c:
             c.argument('default_action', arg_type=get_enum_type(DefaultAction),
                        help='Default action to apply when no rule matches. Only applicable to Premium SKU.')
-            c.argument('public_network_enabled', get_three_state_flag(), help="Allow public network access for the container registry. The Default is allowed")
+            c.argument('public_network_enabled', get_three_state_flag(), help="Allow public network access for the container registry. The Default is to allow public access.")
+            c.argument('allow_trusted_services', get_three_state_flag(), is_preview=True, help="Allow trusted Azure Services to access network restricted registries. For more information, please visit https://aka.ms/acr/trusted-services. The default is to allow.")
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:86]: Help message says `The default is to allow`.

For `acr update`, there should be no default values for the options in update commands.
For `acr create`, I don't see the logic to default to allow. Is it handled on service side?

Comment (adewaleo) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:86]: yes, this is the default on the service side. Should I remove this?

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:86]: You can keep `The default is to allow` for `acr create`, but should remove it for `acr update`. If `allow_trusted_services` is none in update command, then the setting will not be changed.

Comment (adewaleo) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:86]: yes this makes sense. I see what you mean. Thanks for the clarification

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**